sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/HttpTransportClient.java

```diff
@@ -108,7 +108,7 @@ public void close() {
 
             ResourceOperation resourceOperation = new ResourceOperation(request.getOperationType(), request.getResourceType());
             // uuid correlation manager
-            String activityId = request.getActivityId();
+            String activityId = request.getActivityId().toString();
```

Comment:
Possible that in future there might be more string serilization consumers. Is it worth request having both byte[] or string part of Request?

Reply:
This is in my list to explore.
Libraries like avro have similar optimization:
https://github.com/rdblue/avro-java/blob/master/avro/src/main/java/org/apache/avro/util/Utf8.java